### PR TITLE
gradient of InterpolatedTensorProductGridData

### PR DESCRIPTION
**Pull request by danshapero**
This PR adds an implementation of the `gradient` method for the `InterpolatedTensorProductGridData` class and an accompanying test.

### CODE REVIEW

File: include/deal.II/base/function_lib.h [hunk 1]
```diff
@@ -1154,6 +1154,22 @@ namespace Functions
...
+    /**
+     * Compute the gradient of the function defined by bilinear interpolation
+     * of the given data set.
+     *
+     * @param p The point at which the function gradient is to be evaluated.
+     * @param component The vector component. Since this function is scalar,
+     * only zero is a valid argument here.
+     * @return The value of the gradient of the interpolated function at this
+     * point. If the point lies outside the set of coordinates, the function
+     * is extend by a constant and so its gradient is extended by 0.
```
Comment (bangerth) [inline include/deal.II/base/function_lib.h]: extend -> extended

**Comment by bangerth** [comment]
Very nice! Can you fix the one typo and add an entry to `doc/news/changes.h` in the "general" section?

**Comment by danshapero** [comment]
Done and rebased!

**Comment by bangerth** [comment]
Great, thanks very much!

**Comment by tamiko** [comment]
@bangerth I will revert this pull request on the ```dealii-8.3``` branch, and reset the history to the release commit. After that I manually apply it to current master.

**Comment by bangerth** [comment]
Oh, darn. None of us apparently noticed this :-( Thanks for seeing the problem!

**Comment by tamiko** [comment]
Found by accident while helping a colleague.

**Comment by tamiko** [comment]
Successfully applied to master. We can pretend this never happened. :-)

**Comment by danshapero** [comment]
Woops, sorry about this!

**Pull request merged**